IPython/extensions/autoreload.py

```diff
@@ -552,21 +605,45 @@ def autoreload(self, parameter_s=""):
           autoreloaded.
 
         """
-        if parameter_s == "":
+        args = magic_arguments.parse_argstring(self.autoreload, line)
+        mode = args.mode.lower()
+
+        def p(msg):
+            print(msg)
+
+        def l(msg):
+            logging.getLogger("autoreload").info(msg)
```

Comment:
Unless there is a case for hot-swapping the logger at execution.

```suggestion
        logger = logging.getLogger("autoreload")
        def l(msg):
            logger.info(msg)
```